internal/generateFlatbuffer.ts

```diff
@@ -0,0 +1,137 @@
+import { FoxgloveMessageField, FoxglovePrimitive, FoxgloveSchema } from "./types";
+
+// Flatbuffer only supports nested vectors via table
+export const BYTE_VECTOR_FB = `table ByteVectorForNesting {
+  data:[byte];
+}
+
+root_type ByteVectorForNesting;
+`;
+
+// Same as protobuf wellknown types
+export const TIME_FB = `struct Time {
+  sec:long;
+  nsec:int;
+}
+`;
+
+export const DURATION_FB = `struct Duration {
+  sec:long;
+  nsec:int;
+}
+`;
+
+// fields that would benefit from having a default of 1
+const defaultOneNumberFields = new Set(["x", "y", "z", "r", "g", "b", "a", "w"]);
+function isDefaultOneField(field: FoxgloveMessageField): boolean {
+  return (
+    field.type.type === "primitive" &&
+    field.type.name === "float64" &&
+    defaultOneNumberFields.has(field.name)
+  );
+}
```

Comment:
This doesn't belong here because it's not really specific to flatbuffers. If we want to add defaults to some fields, we need to add the default values to `schemas.ts` and just read them here.

Reply:
Added to the schema under `defaultValue?: string` if that seems appropriate.